Added table to show Kubernetes options (#1393)
dapr/docs: daprdocs/content/en/operations/components/setup-secret-store/supported-secret-stores/azure-keyvault.md

    value: "[your_service_principal_tenant_id]"
  - name: spnClientId
    value: "[your_service_principal_app_id]"
  - name: spnCertificate

>     value : "[pfx_certificate_contents_k8s_secret_name]"

Reviewer: Is `spnCertificate` specific to Kubernetes, or could it be used on other platforms given the certificate data is provided in the correct format? If so I'd say:
```suggestion
    value : "[pfx_certificate_contents]"
```

| spnCertificateFile | Y        | PFX certificate file path. <br></br> For Windows the `[pfx_certificate_file_fully_qualified_local_path]` value must use escaped backslashes, i.e. double backslashes. For example `"C:\\folder1\\folder2\\certfile.pfx"`. <br></br> For Linux you can use single slashes. For example `"/folder1/folder2/certfile.pfx"`.  <br></br> See [configure the component](#configure-the-component) for more details | `"C:\\folder1\\folder2\\certfile.pfx"`, `"/folder1/folder2/certfile.pfx"`        |

### Kubernetes

| Field              | Required | Details                                                                 | Example                  |
|--------------------|:--------:|-------------------------------------------------------------------------|--------------------------|
| vaultName          | Y        | The name of the Azure Key Vault                                         | `"mykeyvault"`           |
| spnTenantId        | Y        | Service Principal Tenant Id                                        | `"spnTenantId"`          |
| spnClientId        | Y        | Service Principal App Id                                           | `"spnAppId"`             |

> | spnCertificate     | Y        | Store the certificate for the service principal into the Kubernetes Secret Store.<br></br> See [configure the component](#configure-the-component) for more details | kubectl create secret generic [name] --from-file=[fully_qualified_local_path]        |

Reviewer: Love the instructions on how to create and store the secret, but the command to get the secret in the store should be in the Description. There should also be instructions here on how to reference the certificate secret once it's in the store. The Example should be what to place in the spec yaml file instead of how to get it there.

Author: I agree the document could be more complete with the steps on how to store the file as a secret in K8s. I will work on that this week.

Author: This was already in the table 

```
kubectl create secret generic [name] –from-file=[fully_qualified_local_path]
```
Do you want more than that?

> | spnCertificate     | Y        | Store the certificate for the service principal into the Kubernetes Secret Store.<br></br> See [configure the component](#configure-the-component) for more details | kubectl create secret generic [name] --from-file=[fully_qualified_local_path]        |

Reviewer: Also did we make a change to the component so that `spnCertificate` is a valid field? If so `spnCertificateFile` wouldn't be required.

Author: Both spnCertificate and spnCertificateFile have always been there. But unless you clicked on the Kubernetes tab at the bottom of the page there was no way to know that spnCertificate was even an option. I was trying to surface the existence of the field sooner in the doc. When I was working with this component last week I was stuck until @msfussell pointed to that tab. I wasted a lot of time not knowing there was another option to get the data to the component.

> ## Spec metadata fields
> 
> ### Self-Hosted
> 
> | Field              | Required | Details                                                                 | Example                  |
> |--------------------|:--------:|-------------------------------------------------------------------------|--------------------------|
> | vaultName          | Y        | The name of the Azure Key Vault                                         | `"mykeyvault"`           |
> | spnTenantId        | Y        | Service Principal Tenant Id                                        | `"spnTenantId"`          |
> | spnClientId        | Y        | Service Principal App Id                                           | `"spnAppId"`             |
> | spnCertificateFile | Y        | PFX certificate file path. <br></br> For Windows the `[pfx_certificate_file_fully_qualified_local_path]` value must use escaped backslashes, i.e. double backslashes. For example `"C:\\folder1\\folder2\\certfile.pfx"`. <br></br> For Linux you can use single slashes. For example `"/folder1/folder2/certfile.pfx"`.  <br></br> See [configure the component](#configure-the-component) for more details | `"C:\\folder1\\folder2\\certfile.pfx"`, `"/folder1/folder2/certfile.pfx"`        |
> 
> 
> ### Kubernetes
> 
> | Field              | Required | Details                                                                 | Example                  |
> |--------------------|:--------:|-------------------------------------------------------------------------|--------------------------|
> | vaultName          | Y        | The name of the Azure Key Vault                                         | `"mykeyvault"`           |
> | spnTenantId        | Y        | Service Principal Tenant Id                                        | `"spnTenantId"`          |
> | spnClientId        | Y        | Service Principal App Id                                           | `"spnAppId"`             |
> | spnCertificate     | Y        | Store the certificate for the service principal into the Kubernetes Secret Store.<br></br>  kubectl create secret generic [secretName] --from-file=[secretKey]=[fully_qualified_local_path]<br></br> See [configure the component](#configure-the-component) for more details |   See this guide on [referencing secrets]({{< ref component-secrets.md >}}) to retrieve and use the secret with Dapr components      |
> 

Reviewer: To be consistent with bottom of page let's wrap this in tabs:
```suggestion
## Spec metadata fields

{{< tabs Self-Hosted Kubernetes >}}
{{% codetab %}}

| Field   | Required | Details  | Example  |
|--------|:----------:|----------|-----------|
| vaultName          | Y        | The name of the Azure Key Vault                                         | `"mykeyvault"`           |
| spnTenantId        | Y        | Service Principal Tenant Id                                        | `"spnTenantId"`          |
| spnClientId        | Y        | Service Principal App Id                                           | `"spnAppId"`             |
| spnCertificateFile | Y        | PFX certificate file path. <br></br> For Windows the `[pfx_certificate_file_fully_qualified_local_path]` value must use escaped backslashes, i.e. double backslashes. For example `"C:\\folder1\\folder2\\certfile.pfx"`. <br></br> For Linux you can use single slashes. For example `"/folder1/folder2/certfile.pfx"`.  <br></br> See [configure the component](#configure-the-component) for more details | `"C:\\folder1\\folder2\\certfile.pfx"`, `"/folder1/folder2/certfile.pfx"`        |

{{% /codetab %}}

{{% codetab %}}

| Field   | Required | Details  | Example  |
|--------|:----------:|----------|-----------|
| vaultName        | Y   | The name of the Azure Key Vault   | `"mykeyvault"` |
| spnTenantId      | Y   | Service Principal Tenant Id  | `"spnTenantId"` |
| spnClientId       | Y   | Service Principal App Id   | `"spnAppId"`  |
| spnCertificate   | Y   | The certificate data bytes encoded in PKCS 12.  See [configure the component](#configure-the-component) for more details on referencing this from a Kubernetes secret. |  `secretKeyRef: ....` See [configure the component](#configure-the-component) and [referencing secrets]({{< ref component-secrets.md >}}) for more information.    |

{{% /codetab %}}

{{< /tabs >}}
```

> ### Self-Hosted
> 
> | Field              | Required | Details                                                                 | Example                  |
> |--------------------|:--------:|-------------------------------------------------------------------------|--------------------------|
> | vaultName          | Y        | The name of the Azure Key Vault                                         | `"mykeyvault"`           |
> | spnTenantId        | Y        | Service Principal Tenant Id                                        | `"spnTenantId"`          |
> | spnClientId        | Y        | Service Principal App Id                                           | `"spnAppId"`             |
> | spnCertificateFile | Y        | PFX certificate file path. <br></br> For Windows the `[pfx_certificate_file_fully_qualified_local_path]` value must use escaped backslashes, i.e. double backslashes. For example `"C:\\folder1\\folder2\\certfile.pfx"`. <br></br> For Linux you can use single slashes. For example `"/folder1/folder2/certfile.pfx"`.  <br></br> See [configure the component](#configure-the-component) for more details | `"C:\\folder1\\folder2\\certfile.pfx"`, `"/folder1/folder2/certfile.pfx"`        |
> 
> 
> ### Kubernetes
> 
> | Field              | Required | Details                                                                 | Example                  |
> |--------------------|:--------:|-------------------------------------------------------------------------|--------------------------|
> | vaultName          | Y        | The name of the Azure Key Vault                                         | `"mykeyvault"`           |
> | spnTenantId        | Y        | Service Principal Tenant Id                                        | `"spnTenantId"`          |
> | spnClientId        | Y        | Service Principal App Id                                           | `"spnAppId"`             |
> | spnCertificate     | Y        | Store the certificate for the service principal into the Kubernetes Secret Store.<br></br>  kubectl create secret generic [secretName] --from-file=[secretKey]=[fully_qualified_local_path]<br></br> See [configure the component](#configure-the-component) for more details |   See this guide on [referencing secrets]({{< ref component-secrets.md >}}) to retrieve and use the secret with Dapr components      |
> 

Reviewer: ```suggestion
### Self-Hosted

| Field              | Required | Details | Example |
|--------------------|:--------:|---------|---------|
| vaultName          | Y        | The name of the Azure Key Vault | `"mykeyvault"`
| spnTenantId        | Y        | Service Principal Tenant Id | `"spnTenantId"`
| spnClientId        | Y        | Service Principal App Id | `"spnAppId"`
| spnCertificateFile | Y        | PFX certificate file path. <br></br> For Windows the `[pfx_certificate_file_fully_qualified_local_path]` value must use escaped backslashes, i.e. double backslashes. For example `"C:\\folder1\\folder2\\certfile.pfx"`. <br></br> For Linux you can use single slashes. For example `"/folder1/folder2/certfile.pfx"`.  <br></br> See [configure the component](#configure-the-component) for more details | `"C:\\folder1\\folder2\\certfile.pfx"`, `"/folder1/folder2/certfile.pfx"`

### Kubernetes

| Field          | Required | Details | Example |
|----------------|:--------:|---------|---------|
| vaultName      | Y        | The name of the Azure Key Vault | `"mykeyvault"`
| spnTenantId    | Y        | Service Principal Tenant Id | `"spnTenantId"`
| spnClientId    | Y        | Service Principal App Id | `"spnAppId"`
| spnCertificate | Y        | PKCS 12 encoded bytes of the certificate. See [configure the component](#configure-the-component) for details on encoding this in a Kubernetes secret. | `secretKeyRef: ...` <br /> See [configure the component](#configure-the-component) for more information.

```
